core/Tracker/Request.php

```diff
@@ -402,7 +413,7 @@ public function getParam($name)
         $paramType = $supportedParams[$name][1];
 
         if ($this->hasParam($name)) {
-            $this->paramsCache[$name] = Common::getRequestVar($name, $paramDefaultValue, $paramType, $this->params);
+            $this->paramsCache[$name] = $this->replaceUnsupportedUtf8Chars(Common::getRequestVar($name, $paramDefaultValue, $paramType, $this->params), $name);
```

Comment:
Not sure about this... is it needed right now? After a request has been sanitized... I wonder if you can craft certain requests that end up being basically "unsanitized" afterwards

Reply:
I've added that as some parameters are sent json_encode'd. Otherwise sending a 4byte char e.g. in ecommerce items (as done in the test) will result in an error. (The `Common::getRequestVar` will make the json_decode)

Reply:
we'll just need to make sure it can ideally not lead to any security issues. In other code it is applied before calling `sanitize` but calling it afterwards doesn't look as safe to me

Reply:
That's the reason why the chars gets replaced with a supported char, to avoid someone sends some XSS content that gets "valid" when the unsupported chars are removed in between...

Reply:
makes sense 👍